src/gbm/gbtree.cc

```diff
@@ -475,14 +509,277 @@ class GBTree : public GradientBooster {
   std::vector<std::unique_ptr<TreeUpdater> > updaters;
 };
 
+// dart
+class Dart : public GBTree {
+ public:
+  Dart() {
+    num_pbuffer = 0;
```

Comment:
the initializations was not necessary. vectors are initially empty. num_pbuffer is initialized by constrcuctor of GBTree, this function can be removed